driver-core/src/main/com/mongodb/internal/connection/CommandHelper.java

```diff
@@ -47,24 +47,23 @@ public final class CommandHelper {
     static final String LEGACY_HELLO_LOWER = LEGACY_HELLO.toLowerCase(Locale.ROOT);
 
     static BsonDocument executeCommand(final String database, final BsonDocument command, final ClusterConnectionMode clusterConnectionMode,
-                                       @Nullable final ServerApi serverApi, final InternalConnection internalConnection) {
-        return sendAndReceive(database, command, clusterConnectionMode, serverApi, internalConnection);
+            @Nullable final ServerApi serverApi, final InternalConnection internalConnection, final OperationContext operationContext) {
+        return sendAndReceive(database, command, clusterConnectionMode, serverApi, internalConnection, operationContext);
     }
 
     static BsonDocument executeCommandWithoutCheckingForFailure(final String database, final BsonDocument command,
-                                                                final ClusterConnectionMode clusterConnectionMode,
-                                                                @Nullable final ServerApi serverApi,
-                                                                final InternalConnection internalConnection) {
+            final ClusterConnectionMode clusterConnectionMode, @Nullable final ServerApi serverApi,
+            final InternalConnection internalConnection, final OperationContext operationContext) {
         try {
-            return sendAndReceive(database, command, clusterConnectionMode, serverApi, internalConnection);
+            return executeCommand(database, command, clusterConnectionMode, serverApi, internalConnection, operationContext);
         } catch (MongoServerException e) {
             return new BsonDocument();
         }
     }
 
     static void executeCommandAsync(final String database, final BsonDocument command, final ClusterConnectionMode clusterConnectionMode,
-                                    @Nullable final ServerApi serverApi, final InternalConnection internalConnection,
-                                    final SingleResultCallback<BsonDocument> callback) {
+            @Nullable final ServerApi serverApi, final InternalConnection internalConnection, final OperationContext operationContext,
+            final SingleResultCallback<BsonDocument> callback) {
```

Comment:
`OperationContext` isn't being used for socket communications here. It seems our unified tests might not be covering the socket interaction thoroughly. Should we consider adding some unit tests to ensure we have comprehensive coverage in this area?

Reply:
Great catch - thanks.

Reply:
Done